src/collapse/collapse-control.tsx

```diff
@@ -0,0 +1,37 @@
+import React, {useMemo, useContext, cloneElement} from 'react';
+
+import dataTests from '../global/data-tests';
+
+import {CollapseContext} from './collapse-context';
+import {COLLAPSE_CONTROL_TEST_ID} from './consts';
+
+type ChildrenFunction = (collapsed: boolean) => React.ReactNode;
+
+type Props = {
+  children: ChildrenFunction | React.ReactNode;
+  'data-test'?: string | null | undefined;
+};
+
+/**
+ * @name CollapseControl
+ */
+
+export const CollapseControl: React.FC<Props> = ({children, 'data-test': dataTest}) => {
+  const {setCollapsed, collapsed, id} = useContext(CollapseContext);
+
+  const child = useMemo<React.ReactElement>(() => {
+    if (typeof children === 'function') {
+      return <p data-test={dataTests(COLLAPSE_CONTROL_TEST_ID, dataTest)}>{children(collapsed)}</p>;
+    }
+
+    return <p data-test={dataTests(COLLAPSE_CONTROL_TEST_ID, dataTest)}>{children}</p>;
+  }, [children, collapsed, dataTest]);
+
+  return cloneElement(child, {
```

Comment:
The current RingUI components do rely heavily on `cloneElement`, but it's worth noting that this API is [considered](https://react.dev/reference/react/cloneElement) legacy and fragile. For that reason, I would recommend [exploring](https://react.dev/reference/react/cloneElement#alternatives) the suggested alternatives when creating a new component.

Reply:
@skoch13 Thank you for mentioning this. Sorry for the delay. I'll create another pull request to suggest improvements for this particular usage of cloneElement.